### PR TITLE
Weapon codex entries

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

adds some codex entries for the lore side of the weapons.

## Why It's Good For The Game

lore

## Changelog
:cl: MJP
add: Codex entries for all your favorite weapons.
/:cl:

### CODE REVIEW

**Comment by Hughgent** [comment]
If you notice any spelling mistakes. use the code review, leave a 'suggestion' with the corrected spellings, and i'll press the green button to make that change.

File: code/modules/projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -288,13 +288,13 @@
 //M4RA marksman rifle
 
 /obj/item/weapon/gun/rifle/m4ra
-	name = "\improper M4RA battle rifle"
-	desc = "The M4RA battle rifle is a designated marksman rifle in service with the TGMC. Only fielded in small numbers, and sporting a bullpup configuration, the M4RA battle rifle is perfect for reconnaissance and fire support teams.\nIt is equipped with rail scope and takes 10x24mm A19 high velocity magazines."
+	name = "\improper T-45 battle rifle"
+	desc ="The T-45 is a light full-powered rifle, thus designating it as a Battle Rifle, it is mostly used by light infantry and scouts. It's designed to be useable at all ranges due to the compact size it is also very adaptable to different situations due to the ability to use specialized ammo. Takes specialized "A19" 10x28mm rounds."
```
Comment (spookydonut) [inline code/modules/projectiles/guns/specialist.dm:292]:
```suggestion
	desc ="The T-45 is a light full-powered rifle, thus designating it as a Battle Rifle, it is mostly used by light infantry and scouts. It's designed to be useable at all ranges due to the compact size it is also very adaptable to different situations due to the ability to use specialized ammo. Takes specialized 'A19' 10x28mm rounds."
```

File: code/modules/projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -331,8 +331,8 @@
 
 //Come get some.
 /obj/item/weapon/gun/smartgun
-	name = "\improper M56B smartgun"
-	desc = "The actual firearm in the 4-piece M56B Smartgun System. Essentially a heavy, mobile machinegun.\nReloading is a cumbersome process requiring a powerpack. Click the powerpack icon in the top left or use special action to reload."
+	name = "\improper T-90A smartgun"
+	desc = "The actual firearm in the 4-piece M56B Smartgun System. This one has TGMC markings, thus designating it a T-90A. It is essentially a heavy, mobile machinegun.\nReloading is a cumbersome process requiring a powerpack. Click the powerpack icon in the top left or use special action to reload."1
```
Comment (spookydonut) [inline code/modules/projectiles/guns/specialist.dm:335]:
```suggestion
	desc = "The actual firearm in the 4-piece M56B Smartgun System. This one has TGMC markings, thus designating it a T-90A. It is essentially a heavy, mobile machinegun.\nReloading is a cumbersome process requiring a powerpack. Click the powerpack icon in the top left or use special action to reload."
```

**Comment by Hughgent** [comment]
This alters the main logic of the `get_codex_entry()` proc. Before, it would check the general list of codex entries and if there was one available, it would present that and be done with it. Now it should work backwards and attempt to get more specific/generated entries before finally checking the list.

**Comment by Hughgent** [comment]
This last push should fix #3791

File: code/controllers/subsystem/codex.dm [hunk 3]
```diff
@@ -42,21 +42,24 @@ SUBSYSTEM_DEF(codex)
 	if(!initialized)
 		return
 	var/searching = "\ref[entry]"
+	if(istype(entry, /atom))
```
Comment (Rohesie) [inline code/controllers/subsystem/codex.dm:45]:
```suggestion
	if(isatom(entry))
```

**Pull request merged**